media/system/js/fields/fielduser.js

```diff
@@ -6,80 +6,82 @@
 /**
  * Field user
  */
-;(function($){
+Joomla = window.Joomla || {};
+
+(function(){
 	'use strict';
 
-	$.fieldUser = function(container, options){
+	Joomla.fieldUser = function(container, options){
 		// Merge options with defaults
-		this.options = $.extend({}, $.fieldUser.defaults, options);
+		this.options = Joomla.extend({}, Joomla.fieldUser.defaults, options);
 
 		// Set up elements
-		this.$container = $(container);
-		this.$modal = this.$container.find(this.options.modal);
-		this.$modalBody = this.$modal.find('.modal-body');
-		this.$input = this.$container.find(this.options.input);
-		this.$inputName = this.$container.find(this.options.inputName);
-		this.$buttonSelect = this.$container.find(this.options.buttonSelect);
+		this.container = document.getElementsByName('container');
+		this.modal = this.document.getElementsByName('container').querySelector(this.options.modal);
+		this.modalBody = this.modal.querySelector('.modal-body');
+		this.input = this.container.querySelector(this.options.input);
+		this.inputName = this.container.querySelector(this.options.inputName);
+		this.buttonSelect = this.container.querySelector(this.options.buttonSelect);
 
 		// Bind events
-		this.$buttonSelect.on('click', this.modalOpen.bind(this));
-		this.$modal.on('hide', this.removeIframe.bind(this));
+		this.buttonSelect.addEventListener('click', this.modalOpen.bind(this));
+		this.modal.addEventListener('hide', this.removeIframe.bind(this));
 
 		// Check for onchange callback,
-		var onchangeStr =  this.$input.attr('data-onchange'), onchangeCallback;
+		var onchangeStr =  this.input.setAttribute('data-onchange'), onchangeCallback;
 		if(onchangeStr) {
 			onchangeCallback = new Function(onchangeStr);
-			this.$input.on('change', onchangeCallback.bind(this.$input));
+			this.input.addEventListener('change', onchangeCallback.bind(this.input));
 		}
 
 	};
 
 	// display modal for select the file
-	$.fieldUser.prototype.modalOpen = function() {
-		var $iframe = $('<iframe>', {
+	Joomla.fieldUser.prototype.modalOpen = function() {
+		var iframe = document.getElementById('<iframe>', {
 			name: 'field-user-modal',
-			src: this.options.url.replace('{field-user-id}', this.$input.attr('id')),
+			src: this.options.url.replace('{field-user-id}', this.input.attr('id')),
```

Comment:
Is this.input.attr('id') works with Vanilla JS?

Reply:
Nope it should be `this.input.getAttribute('id')` . Corrected the issue